src/CodeStyle/CSharp/Tests/FormattingAnalyzerTests.cs

```diff
@@ -50,6 +50,49 @@ void MyMethod()
             await Verify.VerifyCodeFixAsync(testCode, fixedCode);
         }
 
+        [Fact]
+        public async Task TestNeedsIndentationButSuppressed()
+        {
+            var testCode = @"
+class MyClass
+{
+  $$void MyMethod1()
+  $${
+  $$}
+
+#pragma warning disable format
+		void MyMethod2()
+  {
+  }
+#pragma warning restore format
+
+  void MyMethod3()
+  $${
+  $$}
+}
+";
+            var fixedCode = @"
+class MyClass
+{
+    void MyMethod1()
+    {
+    }
+
+#pragma warning disable format
+		void MyMethod2()
+  {
+  }
+#pragma warning restore format
+
+  void MyMethod3()
```

Comment:
ya. one of improvement I wanted was changing the formatter not to stop at token but flatten out token and trivia and let rule to specify rule between token and trivia.

basically, rather than  use Roslyn's view of what is token and trivia, use the view of whitespace vs anything else and let one to define rule between "non whitespace token/triva"